### ISSUE TITLE
Android Gradle Plugin 4.0.0 Incompatible 

### ISSUE THREAD

**Issue by eyedol** [Archinamon/android-gradle-aspectj#110]
The current 4.0.0 beta releases eg. `4.0.0-beta04` of AGP isn't compatible with `v4.1.0` of this plugin. 
When AGP is upgraded to say `4.0.0-beta04`  it throws this error

```
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':analytics'.
> Failed to notify project evaluation listener.
   > com.android.build.gradle.internal.variant.BaseVariantData.getVariantConfiguration()Lcom/android/build/gradle/internal/core/GradleVariantConfiguration;
   > com.android.build.gradle.internal.variant.BaseVariantData.getVariantConfiguration()Lcom/android/build/gradle/internal/core/GradleVariantConfiguration;

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

AGP 4+ has removed internal APIs that is used by this plugin

**Comment by davidwong** [comment]
Now that Android Studio 4.0 is released, have the same issue with AGP 4.0.0.

**Comment by Archinamon** [comment]
Released: 4.2.1
But it will not be compatible with earlier versions of AGP :(
